### PR TITLE
Add parsing support for ASSIGN_SHIFT_RIGHT token (>>= operator)

### LINKED ISSUE #82: Add Parsing for `Assignment Shift Right`  Token
Update the parser to handle the ASSIGN_SHIFT_RIGHT token, allowing it to correctly parse expressions involving the >>= operator.

### PR DESCRIPTION
**Pull request by anshikavashistha**
This PR addresses issue #82 by adding support for parsing the ASSIGN_SHIFT_RIGHT token (>>= operator) in the translator parser. This enhancement ensures that the parser correctly recognizes and processes expressions involving the >>= operator.
Closes #82

### CODE REVIEW

**Comment by anshikavashistha** [comment]
@samthakur587 @NripeshN PTAL

**Review by samthakur587 — COMMENTED** [review]
hii @anshikavashistha  can you please solve the merge conflicts and assign issue your self before making the PR 😄  

i have assigned you for this issue .

File: crosstl/src/translator/lexer.py [hunk 1]
```diff
@@ -48,6 +48,7 @@
     ("ASSIGN_AND", r"&="),
     ("ASSIGN_OR", r"\|="),
     ("ASSIGN_XOR", r"\^="),
+    ("ASSIGN_SHIFT_RIGHT", r">>="),
     ("LOGICAL_AND", r"&&"),
```
Comment (samthakur587) [inline crosstl/src/translator/lexer.py:52]: hii @anshikavashistha this task is required to changes in parser.py file at translator file which is located at `crosstl/src/translator/parser.py` . you can pull the latest changes the Token `ASSIGN_SHIFT_RIGHT` is already added .

File: tests/test_translator/test_parser.py [hunk 2]
```diff
@@ -207,3 +207,16 @@ def test_function_call():
...
+    code = """
+    int x = 16;
+    x >>= 2;
+    """
+
+    try:
+        tokens = tokenize_code(code)
+        parse_code(tokens)
+    except SyntaxError as e:
+        pytest.fail(f"Failed to parse ASSIGN_SHIFT_RIGHT token: {e}")
```
Comment (samthakur587) [inline tests/test_translator/test_parser.py:222]: the test is currently not passing bcz you havn't made any changes in parser.py file.after adding support for this Token at parser.py file then this test will pass.

**Review by samthakur587 — CHANGES_REQUESTED** [review]
hii @anshikavashistha can you make these changes i have requested .

**Comment by samthakur587** [comment]
hii @anshikavashistha can you please resolve the merge conflicts

**Review by samthakur587 — APPROVED** [review]
hii @anshikavashistha changes LGTM 🚀

**Pull request merged**